components/project-list/project-list.jsx

```diff
@@ -16,28 +16,48 @@ export default React.createClass({
       displayBatchIndex: 1
     };
   },
+  getDefaultProps() {
+    return {
+      params: {
+        featured: null,
+        issue: null,
+        search: null,
+        ids: null
+      }
+    };
+  },
   componentDidMount() {
     this.fetchData(this.props.params);
   },
   componentWillReceiveProps(nextProps) {
+    // Reset state
+    this.resetState();
```

Comment:
I don't see the problem. I don't think we should reset the state as soon as a user start to type (it should continue to show results for "web" as you add "literacy"). It's ok to have a slight delay. Also don't think we even need a message or a spinner. Message is definitely too quick to read, so maybe a spinner but let's try having nothing and see if users complain.